File: 프로그래머스/2/42578. 의상/의상.js
function solution(clothes) {
    let result = 0;
    const obj = {};
    
    for (let val of clothes) {
        if (!obj[val[1]]) {
            obj[val[1]] = [val[0]];
        } else {
            obj[val[1]].push(val[0]);
        }
    }
    
    for (const key in obj) {
        if (result === 0) {
            result = obj[key].length + 1;
        } else {
            result *= obj[key].length + 1;
        }
    }
    
    return result - 1;
}

File: 프로그래머스/2/42578. 의상/의상.py
def solution(clothes):
    result = 1
    clth_dict = {}
    for c in clothes:
        if clth_dict.get(c[1]) == None:
            clth_dict[c[1]] = []
        clth_dict[c[1]].append(c[0])
    
    for _,value in clth_dict.items():
        result *= len(value) + 1
    
    return result - 1